packages/ra-ui-materialui/src/button/CloneButton.tsx

```diff
@@ -57,13 +56,14 @@ CloneButton.propTypes = {
     record: PropTypes.any,
 };
 
-const enhance = shouldUpdate(
+const PureCloneButton = memo(
+    CloneButton,
     (props: Props, nextProps: Props) =>
-        (props.record &&
-            nextProps.record &&
-            props.record !== nextProps.record) ||
-        props.basePath !== nextProps.basePath ||
-        (props.record == null && nextProps.record != null)
+        (!props.record ||
```

Comment:
This is hard to read. Besides, the initial condition looks dubious. It would probably be better with:

```
(props.record === nextProps.record) &&
(props.basePath === nextProps.basePath)
```

But this lacks the `label` and the `icon`...

In fact, you can omit the second parameter to `memo` completely. I don't understand this optimization (even though I made it myself). We'll re-add it if a profiling session proves that it's useful.